spec/lib/archive_importer_spec.rb

```diff
@@ -138,14 +139,24 @@
           "show_community_spotlight_in_stream" => false,
           "language"                           => "ru",
           "disable_mail"                       => false,
-          "auto_follow_back"                   => true
+          "auto_follow_back"                   => true,
+          "blocks"                             => ["bad_guy@unknown.pod", person_to_block.diaspora_handle]
         }
       }
     }
 
     it "creates user" do
+      # an unknown blocked person
+      stub_request(:get, "https://unknown.pod/.well-known/webfinger?resource=acct:bad_guy@unknown.pod")
+        .to_return(status: 404, body: "", headers: {})
+      stub_request(:get, "https://unknown.pod/.well-known/host-meta")
+        .to_return(status: 404, body: "", headers: {})
+      stub_request(:get, "http://unknown.pod/.well-known/host-meta")
+        .to_return(status: 404)
```

Comment:
Instead of mocking the separate requests (which URLs are called in which order is part of the internals of the federation gem, and the old `host-meta` fallback will eventually be removed ... I know that other places like this already exist in other tests, which is probably where you copied that from, but still less to cleanup later), you maybe want to just make the discovery to directly raise a `DiscoveryError` (which is part of the public API of the federation gem). This is pretty easy if you don't care about which person is tried to be discovered and just want to make everything fail (which works when you move this to a `block_importer_spec` and only care about a single block per test, see my other comment about that). You can look about how this is already done here:

https://github.com/diaspora/diaspora/blob/671e8476ba30b78fa0478955bf2b41a03c26de9d/spec/lib/archive_validator/contact_validator_spec.rb#L59-L62

Reply:
Thats cool - I saw it already somewhere.. but thanks for this snippet!

Reply:
This only works that way, when you only want to fetch and fail a single user, because it just makes all user fetches fail. That's also why I mentioned this should be moved to a `BlockImporter` spec, where each case is tested separately. If you would want to also test a successful fetch, this would fail here.

Reply:
(Hm.. a successful fetch means a 'real' http call.. Is this allowed during tests, did you mean this? ) But I'll see the need to move it to a separate class.

Reply:
You can mock a successful fetch without a http call in a similar way:

https://github.com/diaspora/diaspora/blob/671e8476ba30b78fa0478955bf2b41a03c26de9d/spec/lib/archive_validator/contact_validator_spec.rb#L49-L51

(but this conflicts with the first mocking, so it also only works with a single fetch, which isn't a problem when you test each case separately)